src/fluree/db/query/fql/parse.cljc

```diff
@@ -493,16 +493,32 @@
   [s-mch attrs vars context]
   (trampoline parse-statements* s-mch attrs vars context))
 
+(defn specified-properties?
+  [attrs]
+  (every? v/specified-value? (keys attrs)))
+
+(defn variable-objects?
+  [attrs]
+  (every? v/query-variable? (vals attrs)))
+
+(defn simple-property-join?
+  [id attrs]
+  (and (>= (count attrs) 2)
+       (v/query-variable? id)
+       (specified-properties? attrs)
+       (variable-objects? attrs)))
+
 (defn parse-id-map-pattern
   [m vars context]
-  (let [s-mch (-> m
-                  (get const/iri-id)
-                  (parse-subject context))
+  (let [id    (get m const/iri-id)
+        s-mch (parse-subject id context)
         attrs (dissoc m const/iri-id)]
     (if (empty? attrs)
       [(where/->pattern :id s-mch)]
       (let [statements (parse-statements s-mch attrs vars context)]
-        (sort optimize/compare-triples statements)))))
+        (if (simple-property-join? id attrs)
```

Comment:
Since this optimization only applies to node maps with multiple properties, do we plan to do a pre-parse step of merging node patterns with the same subject value into one map? Right now our SPARQL parser will never produce node maps that satisfy this requirement.

Reply:
Yes, this is one component of many for query optimization. We need to build on the previous pr related to sorting triples by the fewest unspecified variables to also take into account variables that have been resolved in previous triples in the query. Then we can merge contiguous triples from the sorted list with the same subject at that point.

Your point about the SPARQL parser is valid though, and it probably bumps up the priority of including that merging in the optimization.